### PR TITLE
Update README

### PR DESCRIPTION
**Pull request by goosemania**
closes #4236
https://pulp.plan.io/issues/4236

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -1,12 +1,38 @@
-**Acquire, Organize, and Distribute Software**
+**Pulp Plugin API**
+
+The Pulp Plugin API is an essential part of the `Pulp Project 3.0+
+ <https://pypi.python.org/pypi/pulpcore/>`__Pulp 3.0 project which provides a set of base
```
Comment (dkliban) [inline README.rst]: I don't think this link is rendering right.

File: README.rst [hunk 2]
```diff
@@ -1,12 +1,38 @@
...
+classes that can be implemented in plugins to manage content in a way that is consistent 
+across plugins, while still allowing plugin writers the freedom to define their workflows
+as they deem necessary.
+
+The Pulp Plugin API allows you, as a Plugin Writer:
```
Comment (dkliban) [inline README.rst]: s/allows you, as a Plugin Writer/allows plugin writers/

File: README.rst (hunk 1 above)
Comment (asmacdo) [inline README.rst]: This line doesn't render correctly. https://github.com/pulp/pulpcore-plugin/blob/8c82e56ef43bd37e3cff2faedad168111b8f2a1b/README.rst

File: README.rst [hunk 3]
```diff
@@ -1,12 +1,38 @@
...
+ - to easily define your content types, and CRUD operations for all the base parts,
+ - to use high-performance asynchronous download machinery,
```
Comment (asmacdo) [inline README.rst]: s/download/task/

File: README.rst (hunk 1 above)
Comment (goosemania) [reply README.rst]: updated

**Pull request merged**